### ISSUE TITLE
Missing PropertyChanged for MapControl

### ISSUE THREAD

**Issue by charlenni** [Mapsui/Mapsui#532]
The title says it: I'm missing a PropertyChanged event for MapControl.

I want to create default layers if a new Map is set for a MapControl. But I'm having no event for this. So it would be useful to have a PropertyChanged event for MapControl. 

For this
````
public partial class MapControl : INotifyPropertyChanged
````
and
````
#if __FORMS__
        public new event PropertyChangedEventHandler PropertyChanged;

        protected override void OnPropertyChanged([CallerMemberName] string propertyName = "")
        {
            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
        }
#else
        public event PropertyChangedEventHandler PropertyChanged;

        protected void OnPropertyChanged([CallerMemberName] string propertyName = "")
        {
            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
        }
#endif
````
would help. Than this function should be called in the setter of Map and all other setters.

**Comment by pauldendulk** [comment]
There is also Fody.PropertyChanged https://github.com/Fody/PropertyChanged

I don't think I want to add this as another dependency but perhaps it is not necessary to make the Mapsui package depende on Fody.PropertyChanged. I would think this interleaving is done during compile.

**Comment by charlenni** [comment]
You are right. I don’t think, that we need a library for this. I made a PR, where all changes are made.

**Comment by pauldendulk** [comment]
What I was trying to say is that we should use a library if the nuget package does not need a dependency on Fody.PropertyChanged. I think that might be the case because Fody uses interweaving, a technique that is applied when building the dlls. 

But we can investigate this later on.

**Comment by charlenni** [comment]
Solved by #537.
